Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanPsatdAlgorithmRZ.cpp

```diff
@@ -0,0 +1,356 @@
+                                             const std::unique_ptr<amrex::MultiFab>& rho )
+{
+    // Profiling
+    WARPX_PROFILE( "GalileanPsatdAlgorithmRZ::CurrentCorrection" );
+
+    using Idx = SpectralFieldIndex;
+
+    // Forward Fourier transform of J and rho
+    field_data.ForwardTransform( *current[0], Idx::Jx,
+                                 *current[1], Idx::Jy);
+    field_data.ForwardTransform( *current[2], Idx::Jz, 0);
+    field_data.ForwardTransform( *rho, Idx::rho_old, 0 );
+    field_data.ForwardTransform( *rho, Idx::rho_new, 1 );
+
+    // Loop over boxes
+    for (amrex::MFIter mfi(field_data.fields); mfi.isValid(); ++mfi){
+
+        amrex::Box const & bx = field_data.fields[mfi].box();
+
+        // Extract arrays for the fields to be updated
+        amrex::Array4<Complex> fields = field_data.fields[mfi].array();
+
+        // Extract pointers for the k vectors
+        auto const & kr_modes = field_data.getKrArray(mfi);
+        amrex::Real const* kr_arr = kr_modes.dataPtr();
+        amrex::Real const* modified_kz_arr = modified_kz_vec[mfi].dataPtr();
+        int const nr = bx.length(0);
+
+        // Local copy of member variables before GPU loop
+        amrex::Real vz = m_v_galilean[2];
+        amrex::Real const dt = m_dt;
+
+        // Loop over indices within one box
+        int const modes = field_data.n_rz_azimuthal_modes;
+        ParallelFor(bx, modes,
+        [=] AMREX_GPU_DEVICE(int i, int j, int k, int mode) noexcept
+        {
+
+            // All of the fields of each mode are grouped together
+            using Idx = SpectralFieldIndex;
+            auto const Jp_m = Idx::Jx + Idx::n_fields*mode;
+            auto const Jm_m = Idx::Jy + Idx::n_fields*mode;
+            auto const Jz_m = Idx::Jz + Idx::n_fields*mode;
+            auto const rho_old_m = Idx::rho_old + Idx::n_fields*mode;
+            auto const rho_new_m = Idx::rho_new + Idx::n_fields*mode;
+
+            // Shortcuts for the values of J and rho
+            Complex const Jp = fields(i,j,k,Jp_m);
+            Complex const Jm = fields(i,j,k,Jm_m);
+            Complex const Jz = fields(i,j,k,Jz_m);
+            Complex const rho_old = fields(i,j,k,rho_old_m);
+            Complex const rho_new = fields(i,j,k,rho_new_m);
+
+            // k vector values, and coefficients
+            // The k values for each mode are grouped together
+            int const ir = i + nr*mode;
+            amrex::Real const kr = kr_arr[ir];
+            amrex::Real const kz = modified_kz_arr[j];
+            amrex::Real const k_norm2 = kr*kr + kz*kz;
+
+            constexpr Complex I = Complex{0._rt,1._rt};
+
+            // Correct J
+            if ( k_norm2 != 0 )
+            {
+                Complex const theta2 = amrex::exp(I*kz*vz*dt);
+                Complex const inv_1_T2 = 1._rt/(kz*vz == 0 ?  1._rt : 1._rt - theta2);
+                Complex const j_corr_coef = (kz == 0 ? 1._rt/dt : I*kz*vz*inv_1_T2);
```

Comment:
By looking at equations (C1)-(C2) of ([Lehe et al, PRE, 2016](https://journals.aps.org/pre/abstract/10.1103/PhysRevE.94.053305)), it seems to me that we should have a minus sign here. Could you crosscheck?
```suggestion
                Complex const j_corr_coef = (kz == 0 ? 1._rt/dt : -I*kz*vz*inv_1_T2);
```

Reply:
Comparing to what is in fbpic, the minus sign is correct. Thanks for checking! I'll commit this change.